managers/docker-manager/DockerService.js

```diff
@@ -0,0 +1,801 @@
+  //           title: 'Standard',
+  //           language: 'ansi',
+  //           content: logs[0]
+  //         }, {
+  //           id: 'stderr',
+  //           title: 'Error',
+  //           language: 'xxxx',
+  //           content: logs[1]
+  //         }]
+  //       };
+  //     });
+  // }
+
+  // getContainerStatus(state) {
+  //   if (state.Running) {
+  //     return `Up for ${utils.getTimeAgo(state.StartedAt, true)}${state.Paused ? ' (Paused)' : ''}`;
+  //   }
+  //   if (state.ExitCode > 0) {
+  //     return `Exited (${state.ExitCode}) ${utils.getTimeAgo(state.FinishedAt)}`;
+  //   }
+  //   return 'Stopped';
+  // }
+
+  getEnvironment() {
+    return _
+      .chain(this.containerInfo)
+      .get('Config.Env')
+      .map(kv => _.slice(/^([^=]+)=(.*)$/.exec(kv), 1))
+      .fromPairs()
+      .value();
+  }
+
+  // getDetails() {
+  //   logger.info(`Building details hash for container '${this.containerName}'...`);
+  //   const info = {
+  //     'ID': this.containerInfo.Id,
+  //     'Name': this.containerInfo.Name,
+  //     'Created': utils.getTimeAgo(this.containerInfo.Created),
+  //     'Status': this.getContainerStatus(this.containerInfo.State)
+  //   };
+  //   const config = {
+  //     'Image': this.containerInfo.Config.Image,
+  //     'Entrypoint': _.join(this.containerInfo.Config.Entrypoint, ' '),
+  //     'Command': _.join(this.containerInfo.Config.Cmd, ' '),
+  //     'Work Directory': this.containerInfo.Config.WorkingDir,
+  //     'User': this.containerInfo.Config.User
+  //   };
+  //   const hostConfig = {
+  //     'CPU Shares': this.containerInfo.HostConfig.CpuShares,
+  //     'Memory': this.containerInfo.HostConfig.Memory,
+  //     'Memory Swap': this.containerInfo.HostConfig.MemorySwap,
+  //   };
+  //   const networkSettings = {
+  //     'Host IP': undefined,
+  //     'Container IP': this.containerInfo.NetworkSettings.IPAddress
+  //   };
+  //   const details = {
+  //     'Container': info,
+  //     'Configuration': config,
+  //     'Host Configuration': hostConfig,
+  //     'Environment Variables': this.getEnvironment()
+  //   };
+  //   if (this.containerInfo.State.Running) {
+  //     _.assign(hostConfig, {
+  //       'Privileged': this.containerInfo.HostConfig.Privileged,
+  //     });
+  //     const networkInfo = this.getNetworkInfo(this.containerInfo);
+  //     _.assign(networkSettings, {
+  //       'Host IP': networkInfo.ip
+  //     });
+  //     _.assign(details, {
+  //       'Network Settings': networkSettings,
+  //       'Exposed Ports': networkInfo.ports,
+  //       'Exposed Volumes': this.containerInfo.HostConfig.Binds
+  //     });
+  //   }
+  //   return details;
+  // }
+
+  static createDockerService(instanceId, options) {
+    const planId = options.plan_id;
+    const plan = catalog.getPlan(planId);
+    const context = _.get(options, 'context');
+    const dockerService = new DockerService(instanceId, plan);
+
+    return DockerImageLoaderService.load(plan)
```

Comment:
earlier on start of server the docker image was getting loaded, now on demand during creation they are loading. Since the operation of docker creation is Synch, this would increase the time required for provisioning and more chances of create timing out and orphaned containers. Shouldn't this be done at bootstrap?

Reply:
No, earlier also it was happening during the load of DockerManager. The code is same as DockerManager.load();